components/arm/fake/fake.go

```diff
@@ -24,25 +22,47 @@ import (
 	"go.viam.com/rdk/spatialmath"
 )
 
+// errAttrCfgPopulation is the returned string if the AttrConfig's fields are fully populated.
+const errAttrCfgPopulation = "can only populate either ArmModel or ModelPath - not both"
+
 // ModelName is the string used to refer to the fake arm model.
 var ModelName = resource.NewDefaultModel("fake")
 
-//go:embed fake_model.json
-var fakeModelJSON []byte
-
 // AttrConfig is used for converting config attributes.
 type AttrConfig struct {
-	FailNew      bool   `json:"fail_new"`
-	FailValidate bool   `json:"fail_validate"`
-	ArmModel     string `json:"arm-model"`
+	ArmModel      string `json:"arm-model,omitempty"`
+	ModelFilePath string `json:"model-path,omitempty"`
+}
+
+func modelFromName(model, name string) (referenceframe.Model, error) {
+	switch resource.ModelName(model) {
+	case xarm.ModelName6DOF.Name:
+		return xarm.Model(name, 6)
+	case xarm.ModelName7DOF.Name:
+		return xarm.Model(name, 7)
+	case ur.ModelName.Name:
+		return ur.Model(name)
+	case yahboom.ModelName.Name:
+		return yahboom.Model(name)
+	case eva.ModelName.Name:
+		return eva.Model(name)
+	default:
+		return nil, errors.Errorf("fake arm cannot be created, unsupported arm_model: %s", model)
+	}
 }
 
 // Validate ensures all parts of the config are valid.
 func (config *AttrConfig) Validate(path string) error {
-	if config.FailValidate {
-		return errors.New("whoops! failed to validate")
+	var err error
+	switch {
+	case config.ArmModel != "" && config.ModelFilePath != "":
+		return errors.New(errAttrCfgPopulation)
+	case config.ArmModel != "" && config.ModelFilePath == "":
+		_, err = modelFromName(config.ArmModel, "")
+	case config.ArmModel == "" && config.ModelFilePath != "":
+		_, err = referenceframe.ModelFromPath(config.ModelFilePath, "")
 	}
```

Comment:
Shouldn't there be some default case here?

Reply:
ok. default could then only be `config.ArmModel == "" && config.ModelFilePath == ""`. Hence `return nil` for default.

Reply:
I think your default case is the one that is listed first at the moment.  You error on neither or both being specified

Reply:
Shouldn't that default (both being empty string) be an error rather than nil?

Reply:
I am seeing divergence on what we think the default case should be @biotinker @raybjork .

Reply:
I think we are both saying that both or neither should be the default case and should result in an error

Reply:
Which case is the "default" doesn't really matter as long as all four combinations are captured: one populated, the other populated, both, or neither. The first two should try to resolve their values, the latter two should error immediately. Which is what I see happening here; this looks correct.

Reply:
@raybjork are you saying remove the first case and have both fall through to default?

We can do that, but @nfranczak  we still want to distinguish between the "both" and "neither" case errors, so that check would happen within default

Reply:
Side note: this might be more easily read as if/else rather than switch/case/default

Reply:
Oh I thought the error already said "need only one" so I assumed it applied to both.  But yes if you want to distinguish you are correct. You could do this any number of ways I'm fine with whatever

Reply:
I am generally in favor of more specific, granular error messages :)

Reply:
i will leave as is. IMO putting the error cases within the default and then doing nested conditionals is less readable.